client/state/activity-log/log/is-discarded/index.js

```diff
@@ -0,0 +1,52 @@
+/** @format */
+/**
+ * External dependencies
+ */
+
+import { memoize } from 'lodash';
+
+/**
+ * Normalize timestamp values
+ *
+ * Some timestamps are in seconds instead
+ * of in milliseconds and this will make
+ * sure they are all reported in ms
+ *
+ * The chosen comparison date is older than
+ * WordPress so no backups should already
+ * exist prior to that date 😉
+ *
+ * @param {Number} ts timestamp in 's' or 'ms'
+ * @returns {Number} timestamp in 'ms'
+ */
+const ms = ts =>
+	ts < 946702800000 // Jan 1, 2001 @ 00:00:00
+		? ts * 1000 // convert s -> ms
+		: ts;
+
+export const getRewinds = events =>
+	events
+		.filter( ( { activityName } ) => activityName === 'rewind__complete' )
+		.map( ( { activityTs, activityTargetTs } ) => [ activityTs, ms( activityTargetTs ) ] );
+
+export const isDiscardeder = ( rewinds, viewFrom ) => {
+	const isDiscarded = memoize(
+		ts =>
+			ts > viewFrom ||
+			rewinds.some(
+				( [ rp, bp ] ) => ! ( bp >= ts || ts >= rp || rp > viewFrom || isDiscarded( rp ) )
```

Comment:
I think after I add the doc block it will be clearer. I’m open to changing them later, but right now I’ve felt like having the “formula” compact has been much easier to follow and debug if you know what the three variable are

Reply:
Yeah, a doc block that explains what they would be fine 👍